File: Lecture17: Bubble Sort/BubbleSort.c
#include <stdio.h>

void PrintArray(int *A, int n)
{
    int i;
    for (i = 0; i < n; i++)
    {
        printf("%d ", A[i]);
    }
    printf("\n");
}

void BubbleSort(int *A, int n)
{
    int i, j, temp;
    for (i = 0; i < n - 1; i++)
    {
        printf("Working on pass Number:%d\n", i + 1);
        for (j = 0; j < n - 1 - i; j++)
        {
            if (A[j] > A[j + 1])
            {
                temp = A[j];
                A[j] = A[j + 1];
                A[j + 1] = temp;
            }
        }
    }
}
int main()
{
    int A[] = {1, 9, 6, 3, 78, 56};
    int n = 6;
    PrintArray(A, n);
    BubbleSort(A, n);
    PrintArray(A, n);
    return 0;
}

File: Lecture17: Bubble Sort/BubbleSort.java
import java.util.*;
class BubbleSort {
    int[] bubbleSort(int arr[], int n) {
        for(int i = 1 ; i < n; i++) {
            //for round 1 to n-1
            boolean swapped = false;
            
            for(int j = 0 ; j < n-i ; j++) {
                
                //process element till n-i th index
                if(arr[j] > arr[j+1]) {

                    int temp = arr[j];
                    arr[j] = arr[j+1];
                    arr[j+1] = temp;
                    swapped = true;
                }
                
            }
            
            if(swapped == false) {
                //already sorted
                break;
            }
               
        }
        return arr;
    }
}